org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/antlr/ContentAssistContextFactory.java

```diff
@@ -409,7 +412,20 @@ public String getPrefix(INode prefixNode) {
 		if (prefixNode instanceof ILeafNode) {
 			if (((ILeafNode) prefixNode).isHidden() && prefixNode.getGrammarElement() != null)
 				return "";
-			return getNodeTextUpToCompletionOffset(prefixNode);
+			INode startingNode = prefixNode;
+			// As long as there are leaf nodes with syntax errors one after the other go back until the first one is
+			// found
+			while (startingNode instanceof LeafNodeWithSyntaxError
```

Comment:
Please do not cast to concrete types but check node.hasSyntaxError instead.

Reply:
There is no such method ... or at least I'm not able to find it.